src/permitInfo/calculatePermitInfo.ts

```diff
@@ -0,0 +1,154 @@
+import {getTokenPermitInfo, PermitInfo} from 'alfetopito-permit-utils' // import {getTokenPermitInfo, PermitInfo} from '@cowprotocol/permit-utils' // TODO: maybe make the args nicer?
+import * as path from 'node:path'
+import {ethers} from 'ethers'
+import {readFileSync, writeFileSync} from 'node:fs'
+import {JsonRpcProvider} from '@ethersproject/providers'
+
+// TODO: maybe make the args nicer?
+// Get args from cli: chainId, optional token lists path, optional rpcUrl
+const [, , chainId, tokenListPath, rpcUrl] = process.argv
+
+const BASE_PATH = path.join('src/public/')
+
+async function calculatePermitInfo(
+  chainId: number,
+  tokenListPath: string | undefined,
+  rpcUrl: string | undefined,
+): Promise<void> {
+  // Load existing permitInfo.json file for given chainId
+  const permitInfoPath = path.join(BASE_PATH, `PermitInfo.${chainId}.json`)
+
+  let allPermitInfo: Record<string, PermitInfo> = {}
+
+  // Load existing permitInfo.json file for given chainId if it exists
+  try {
+    allPermitInfo = JSON.parse(readFileSync(permitInfoPath, 'utf8')) as Record<string, PermitInfo>
+  } catch (e) {
+    console.info(`File ${permitInfoPath} not found. It'll be created.`, e)
+  }
+
+  // Build provider instance
+  const provider = getProvider(chainId, rpcUrl)
+
+  // Load tokens info from a token list
+  const tokens = getTokens(chainId, tokenListPath)
+
+  // Create a list of promises to check all tokens
+  const promises = tokens.map((token) => {
+    const existingInfo = allPermitInfo[token.address.toLowerCase()]
+
+    return fetchPermitInfo(chainId, provider, token, existingInfo)
+  })
+
+  // Await for all of them to complete
+  const results = await Promise.allSettled(promises)
+
+  // Iterate over each result
+  results.forEach((result) => {
+    // Ignore failed or the ones where the value is falsy
+    if (result.status === 'fulfilled' && result.value) {
+      const [address, permitInfo] = result.value
+
+      // Store result
+      allPermitInfo[address] = permitInfo
+    }
+  })
+
+  try {
+    writeFileSync(permitInfoPath, JSON.stringify(sortPermitInfo(allPermitInfo), undefined, 2))
+  } catch (e) {
+    console.error(`Failed to write file ${permitInfoPath}`, e)
+  }
+}
+
+function getProvider(chainId: number, rpcUrl: string | undefined): JsonRpcProvider {
+  const rpcEndpoint = rpcUrl ? rpcUrl : DEFAULT_RPC_URLS[chainId]
+
+  if (!rpcEndpoint) {
+    throw new Error(`No RPC found for network ${chainId}`)
+  }
+
+  if (!rpcUrl && (chainId === 1 || chainId === 5) && !process.env.INFURA_API_KEY) {
+    throw new Error(`INFURA_API_KEY is required`)
+  }
+
+  return new ethers.providers.JsonRpcProvider(rpcEndpoint)
+}
+
+const DEFAULT_RPC_URLS: Record<number, string> = {
+  1: 'https://mainnet.infura.io/v3/' + process.env.INFURA_API_KEY,
+  5: 'https://goerli.infura.io/v3/' + process.env.INFURA_API_KEY,
+  100: 'https://rpc.gnosischain.com',
+}
```

Comment:
should this be un a more central place?

Reply:
ideally we allow to define the RPCs (since we don't use infura any more).

Reply:
Yep, will. Still not fully fleshed out.
Also, there's one more optional parameter which is a rpc url that takes precedence over these defaults.
It's there just to make it easier.